ts/utils/dates.ts

```diff
@@ -227,16 +227,13 @@ export const isExpired = (
 };
 
 /**
- * This function returns true or false is the provided expiryDate in format "YYYYMM" is expired or not
+ * This function returns true or false is the provided expiryDate is expired or not
  * @param expiryDate
  */
-export const isExpiredDate = (expiryDate: string): boolean => {
-  const year = +expiryDate.slice(0, 4);
-  const month = +expiryDate.slice(4, 6);
+export const isExpiredDate = (expiryDate: Date): boolean => {
   const now = new Date();
   const nowYearMonth = new Date(now.getFullYear(), now.getMonth() + 1);
```

Comment:
a payment method expiring this exact month will be set as expired, if this is not intended I believe it is because of the `Date.getMonth` method, which returns a zero-indexed month

Reply:
You are right. The `+ 1` here should removed because we already subtract 1 to the month in the `getDateFromExpiryDate` function.
Fixed here, I've also added some tests!
See here https://github.com/pagopa/io-app/pull/5674/commits/2f827763b88ef965e7beee18f9e9337f78a89da9